tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -11330,6 +11343,124 @@ TEST_F(ExecutionTest, QuadAnyAll) {
     WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
 }
 
+TEST_F(ExecutionTest, IsNormalTest) {
+  EnableShaderBasedValidation();
+    WEX::TestExecution::SetVerifyOutput verifySettings(
+      WEX::TestExecution::VerifyOutputSettings::LogOnlyFailures);
+  CComPtr<IStream> pStream;
+  ReadHlslDataIntoNewStream(L"ShaderOpArith.xml", &pStream);
+
+  std::shared_ptr<st::ShaderOpSet> ShaderOpSet =
+      std::make_shared<st::ShaderOpSet>();
+  st::ParseShaderOpSetFromStream(pStream, ShaderOpSet.get());
+  st::ShaderOp *pShaderOp =
+      ShaderOpSet->GetShaderOp("IsNormal");
+  vector<st::ShaderOpRootValue> fallbackRootValues = pShaderOp->RootValues;
+
+  const int expectedResultsSize = 12;
+  // false represents 0, true represents a non-zero value
```

Comment:
What's this comment for?  It doesn't seem to clarify anything.  Perhaps a comment like: "Result of IsNormal operation"?

`isnan`, and thus the `isnormal` replacement, returns bool, which translates to 0 or 1 when converted to int or uint.  So it doesn't make a difference in the form `isnan(...) ? 1 : 0`.  The ternary operation doesn't do anything extra.

There isn't any guide to see what each expected result maps to for the input of the operation here either.  That would actually be helpful (or if the inputs were defined here alongside the outputs).